addons/autocomplete_googleplaces/static/src/js/addressForm.js

```diff
@@ -0,0 +1,84 @@
+/** @odoo-module */
+
+import publicWidget from 'web.public.widget';
+import { DropPrevious } from 'web.concurrency';
+import { debounce } from "@web/core/utils/timing";
+
+
+publicWidget.registry.AddressForm = publicWidget.Widget.extend({
+    selector: '.oe_cart .checkout_autoformat',
+    events: {
+        'input input[name="street"]': '_onChangeStreet',
+        'click .js_autocomplete_result': '_onClickAutocompleteResult'
+    },
+
+    init: function() {
+        this.streetAndNumberInput = document.querySelector('input[name="street"]');
+        this.cityInput = document.querySelector('input[name="city"]');
+        this.zipInput = document.querySelector('input[name="zip"]');
+        this.countrySelect = document.querySelector('select[name="country_id"]');
+        this.stateSelect = document.querySelector('select[name="state_id"]');
+        this.dp = new DropPrevious();
+
+        this._onChangeStreet = debounce(this._onChangeStreet, 200);
+        this._super.apply(this, arguments);
+    },
+
+    _onChangeStreet: function(ev) {
+        this.dp.add(
+            this._rpc({
+                route: '/autocomplete/address',
+                params: {
+                    partial_address: ev.currentTarget.value
+                }
+            })).then((response) => {
+            const inputContainer = ev.currentTarget.parentNode;
+            let dropdown = inputContainer.querySelector('.dropdown-menu');
+            if(dropdown) dropdown.remove();
+            inputContainer.appendChild($(response.template)[0])
+        })
+    },
+
+    _onClickAutocompleteResult: function(ev) {
+        const dropDown = ev.currentTarget.parentNode;
+
+        this._rpc({
+            route: '/autocomplete/address_full',
+            params: {
+                address: ev.currentTarget.innerText,
+                google_place_id: ev.currentTarget.dataset.googlePlaceId
+            }
+        }).then((result) => {
+            if(result.address){ // Don't change this field in case something goes wrong
+                if(result.address.street){
+                    this.streetAndNumberInput.value = result.address.formatted_street_number;
+                }
+                // Text fields, empty if no value in order to avoid the user missing old data.
+                this.zipInput.value = result.address.zip ? result.address.zip : '';
+                this.cityInput.value = result.address.city ? result.address.city : '';
```

Comment:
FYI this is also possible.
```suggestion
                this.zipInput.value = result.address.zip || '';
                this.cityInput.value = result.address.city || '';
```